actions/retest/main.go

```diff
@@ -0,0 +1,169 @@
+	requiredReviewCount = os.Getenv("INPUT_REQUIRED-APPROVE-COUNT")
+	exemptlabel         = os.Getenv("INPUT_EXEMPT-LABEL")
+	requiredlabel       = os.Getenv("INPUT_REQUIRED-LABEL")
+	githubToken         = os.Getenv("GITHUB_TOKEN")
+	owner, repo         = func() (string, string) {
+		if os.Getenv("GITHUB_REPOSITORY") != "" {
+			if len(strings.Split(os.Getenv("GITHUB_REPOSITORY"), "/")) == 2 {
+				return strings.Split(os.Getenv("GITHUB_REPOSITORY"), "/")[0], strings.Split(os.Getenv("GITHUB_REPOSITORY"), "/")[1]
+			}
+
+		}
+		return "", ""
+	}()
+)
+
+func main() {
+	if requiredlabel == "" {
+		log.Fatal("requiredlabel are not set")
+	}
+
+	if githubToken == "" {
+		log.Fatal("GITHUB_TOKEN is not set")
+	}
+
+	if owner == "" || repo == "" {
+		log.Fatal("GITHUB_REPOSITORY is not set")
+	}
+
+	retry, err := strconv.Atoi(retryLimit)
+	if err != nil {
+		log.Fatalf("max-retry %q is not valid", retryLimit)
+	}
+
+	totalRequiredReviews, err := strconv.Atoi(requiredReviewCount)
+	if err != nil {
+		log.Fatalf("required-review-count %q is not valid", requiredReviewCount)
+	}
+
+	ctx := context.Background()
+	ts := oauth2.StaticTokenSource(
+		&oauth2.Token{AccessToken: githubToken},
+	)
+	tc := oauth2.NewClient(ctx, ts)
+	client := github.NewClient(tc)
+	opt := &github.PullRequestListOptions{}
+	req, _, err := client.PullRequests.List(context.TODO(), owner, repo, opt)
+	if err != nil {
+		log.Fatalf("failed to list pull requests %v\n", err)
+	}
+	for _, re := range req {
+		if *re.State == "open" {
+			prNumber := re.GetNumber()
+			log.Printf("PR with ID %d with Title %q is open\n", prNumber, re.GetTitle())
+			for _, r := range re.Labels {
+				// check if label is exempt
+				if strings.EqualFold(exemptlabel, r.GetName()) {
+					continue
+				}
+				// check if label is matching
+				if !strings.EqualFold(requiredlabel, r.GetName()) {
+					continue
+				}
+				log.Printf("checking status for PR %d with label %s", prNumber, r.GetName())
+				rs, _, err := client.Repositories.ListStatuses(context.TODO(), owner, repo, re.GetHead().GetSHA(), &github.ListOptions{})
+				if err != nil {
+					log.Printf("failed to list status %v\n", err)
+					continue
+				}
+
+				rev, _, err := client.PullRequests.ListReviews(context.Background(), owner, repo, prNumber, &github.ListOptions{})
+				if err != nil {
+					log.Printf("failed to list reviews %v\n", err)
+					continue
+				}
+				approvedReviews := 0
+				for _, rv := range rev {
+					if rv.GetState() == "APPROVED" {
+						approvedReviews += 1
+					}
+				}
+				if !(approvedReviews >= totalRequiredReviews) {
+					log.Printf("total approvred reviews for PR %d are %d but required %d", prNumber, approvedReviews, totalRequiredReviews)
+					continue
+				}
+				creq, _, err := client.Issues.ListComments(context.Background(), owner, repo, prNumber, &github.IssueListCommentsOptions{})
+				if err != nil {
+					log.Printf("failed to list comments %v\n", err)
+				}
+
+				for _, r := range rs {
```

Comment:
re-using the `r` variable, this is ugly

Reply:
as using l variable fo label this should be fine.